src/h5web/vis-packs/core/utils.ts

```diff
@@ -157,18 +148,13 @@ function extendEmptyDomain(
   return [value - extension, value + extension];
 }
 
-export function clampBound(val: number, withPositiveMin = false): number {
-  const absVal = Math.abs(val);
-
-  if (withPositiveMin && absVal <= 0) {
-    return Number.MIN_VALUE;
-  }
-
-  if (absVal > Number.MAX_VALUE / 2) {
-    return (Math.sign(val) * Number.MAX_VALUE) / 2; // max domain length is `Number.MAX_VALUE`
-  }
+export function clampBound(
+  val: number,
+  supportedDomain: Domain = [-Number.MAX_VALUE / 2, Number.MAX_VALUE / 2]
+): number {
+  const [supportedMin, supportedMax] = supportedDomain;
 
-  return val;
+  return clamp(val, supportedMin, supportedMax);
```

Comment:
`clampBound` is now a simple `clamp` wrapper with optional `supportedDomain`

Reply:
Very nice. I don't think it's worth having a `default` here, since `supportedDomain` is required in the scales record (which I like because it's explicit).

Reply:
Looking more thoroughly into this, and as I mentioned in another comment, I feel like `MAX_VALUE` and `MIN_VALUE` should be an implementation detail of `clampBound`. Having them outside feels too brittle (e.g. risk of typo in the `supportedDomains` of one scale but not another...)

Moreover, is the clamping to 0 really needed for the square root scale in `extendDomain`? Can `extendDomain` ever generate a min value lower than 0? `extendEmptyDomain` can, but perhaps this can be dealt with there?

If I recall, the clamp to `MIN_VALUE` in log is required because the scale can output 0, but that's a very specific exception. Perhaps it is better to make this exception clear by keeping the code as it is? To me, the clamping code that needs more thinking is in `extendEmptyDomain`, so perhaps my suggestion of moving it to `extendDomain` is actually a bad idea.